internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -113,6 +144,88 @@ func (hr *HypershiftAgentServiceConfigReconciler) getKubeconfigSecret(ctx contex
 	return secret, nil
 }
 
+func (hr *HypershiftAgentServiceConfigReconciler) syncSpokeAgentInstallCRDs(ctx context.Context, spokeClient client.Client) error {
+	// Fetch agent-install CRDs using in-cluster client
+	localCRDs, err := hr.getAgentInstallCRDs(ctx, hr.Client)
+	if err != nil {
+		return err
+	}
+	if len(localCRDs.Items) == 0 {
+		return pkgerror.New("agent-install CRDs are not available")
+	}
+
+	// Ensure local agent-install CRDs exist on the spoke cluster
+	if err := hr.ensureSpokeAgentInstallCRDs(ctx, spokeClient, localCRDs); err != nil {
+		return pkgerror.New("Failed to create agent-install CRDs on spoke cluster")
+	}
+
+	// Delete agent-install CRDs that don't exist locally
+	if err := hr.cleanStaleSpokeAgentInstallCRDs(ctx, spokeClient, localCRDs); err != nil {
+		hr.Log.WithError(err).Warn("Failed to remove stale agent-install CRDs from spoke cluster in namespace")
+		return nil
+	}
+
+	return nil
+}
+
+func (hr *HypershiftAgentServiceConfigReconciler) ensureSpokeAgentInstallCRDs(ctx context.Context, spokeClient client.Client, localCRDs *apiextensionsv1.CustomResourceDefinitionList) error {
+	// Ensure CRDs exist on the spoke cluster
+	for _, item := range localCRDs.Items {
+		crd := item
+		c := crd.DeepCopy()
+		c.ResourceVersion = "" // ResourceVersion should not be set on objects to be created
+		mutate := func() error {
+			c.Spec = crd.Spec
+			c.Annotations = crd.Annotations
+			c.Labels = crd.Labels
+			return nil
+		}
+		result, err1 := controllerutil.CreateOrUpdate(ctx, spokeClient, c, mutate)
```

Comment:
no reason to call it `err1` right? it can be just `err`

Reply:
yep, no need now.